core/Cargo.toml

```diff
@@ -255,6 +255,9 @@ reqwest = { version = "0.12.2", features = [
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 tokio = { version = "1.27", features = ["sync", "io-util"] }
+# TODO: I added this dependency in order to not re-implement the Url::query_pairs function.
+# Is it okay to include it? (we are using that crate in the ofs binary https://github.com/apache/opendal/blob/main/bin/ofs/Cargo.toml#L45
+url = "2.5.4"
```

Comment:
Please use `http::uri` instead. During the development of `core`, there is no need to reference other download stream projects. Simply make the best decision based on the current `core` code.